docs/proposals/excluding-drives.md

```diff
@@ -0,0 +1,108 @@
+# Proposal: Specific drives excluding
+
+Last updated: 07.02.2022
+
+
+## Abstract
+
+CSI Baremetal needs to have API for excluding specific drives from the common storage pool. 
+Excluded drives won't be used for allocating new Volumes or LVGs.
+
+## Background
+
+CSI customers might not want to use some disks for whatever reason:
+- for another application not managed by CSI
+- as a workaround for any disk detection issues in CSI
+
+So in these cases CSI haven't to allocate Volumes for some drives.
+
+## Proposal
+
+### User API
+
+The rules to include/exclude drives based on:
+- Size
+- Type (HDD, SSD, ...)
+- Serial Number
+- PID
+- VID
+```yaml
+drive_rule:
+  size:
+    less_then: 1Gi
+    more_then: 100Mi
+  # include drives with the following parameters only
+  in:
+    type:
+      - HDD
+    SN:
+      - sn1
+      - sn2
+    PID:
+      - ...
+    VID:
+      - ...
+  # exclude drives with the following parameters
+  not_in:
+    type:
+      - SSD
+    SN:
+      - ...
+    PID:
+      - ...
+    VID:
+      - ...
+```
+
+Settings can be set for all nodes in the cluster or for only one node. 
+In the node section user chooses is global rules be applied or not.
+```yaml
+drive_options:
+  global:
+    drive_rule: ...
+  nodes:
+    - name: node1
+      enable_global: true/false
+      drive_rule:
+    - name: ...
+```
+
+#### Pass settings
+Propose to process excluded drives on Discover stage in csi-baremetal-node part. 
+Node container will take drive setting from the ConfigMap created with csi-baremetal-deployment.
+
+Node ConfigMap can be updated in the following ways:
+1. Set global drive rule on CSI installation.
+
+Helm command for `csi-baremetal-driver` should be modified with:
+```yaml
+--set driver.node.global_drive_rule.size.more_then="1Gi" \
+--set driver.node.global_drive_rule.in.SN={sn1, sn2, sn3}
+```
+
+The option to modify rules for specific nodes is not supported due to complex formatting.
+
+2. After installing CSI user could edit a ConfigMap with CSI nodes setting to change included/excluded drives list
```

Comment:
It's not clear to me how to take a look or display current settings of the ConfigMap. @safronovD, can you clarify that, if possible?

Reply:
Sure, updated